src/chains/ethereum/ethereum/src/transaction-pool.ts

```diff
@@ -280,23 +293,51 @@ export default class TransactionPool extends Emittery<{ drain: undefined }> {
         // origin's heap we are executable.
         transactionPlacement = TriageOption.Executable;
       }
+      // we've gotten the account's nonce the synchronous way (by looking at
+      // the pending queue), but we still need to look up the account's balance
+      const { balance } = await this.#blockchain.accounts.getNonceAndBalance(
```

Comment:
In the case that we get the nonce from the pending transaction pool, we still need to fetch the account's balance. I've moved this to be _after_ we find the nonce so that finding the nonce is all synchronous.